core/src/test/java/org/mule/tck/junit4/rule/FreePortFinder.java

```diff
@@ -38,24 +46,42 @@ public synchronized Integer find()
     {
         for (int i = 0; i < portRange; i++)
         {
-            int port = minPortNumber + random.nextInt(portRange);
+            try
+            {
+                int port = minPortNumber + random.nextInt(portRange);
+                String portFile = String.valueOf(port) + LOCK_FILE_EXTENSION;
+                FileChannel channel = FileChannel.open(Paths.get(portFile), CREATE, WRITE);
+                FileLock lock = channel.tryLock();
 
-            if (selectedPorts.contains(port))
+                if (isPortFree(port))
```

Comment:
should check that the lock was acqired (`lock != null`). Ref: https://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileChannel.html#tryLock()